### PR TITLE
Git operator fix logger

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Implement GitClientFactory to create Gitclient with context logger

**Related issue(s)**

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/function_reconcile_gitops_test.go [hunk 1]
```diff
@@ -803,14 +815,18 @@ func Test_stateFnGitCheckSources(t *testing.T) {
 			},
 		}
 
-		m := new(automock.GitOperator)
-		m.On("LastCommit", mock.Anything).Return("", fmt.Errorf("test error")).Once()
+		gitClient := new(automock.GitClient)
+		gitClient.On("LastCommit", mock.Anything).Return("", fmt.Errorf("test error")).Once()
```
Comment (anoipm) [inline components/function-controller/internal/controllers/serverless/function_reconcile_gitops_test.go:819]: We have `Once` constraint in this place, but I can't see related `AssertExpectations`.

**Review by m00g3n — APPROVED** [review]
LGTM

**Pull request merged**